### PR TITLE
Adds the ability for electrification to check if the entity is powered by an APC

### PR DESCRIPTION
**Pull request by vulppine**
## About the PR 

Adds the ability for electrification to check if the entity it's checking is powered by an APC. If so, and if it's enabled, it will electrocute the user as if RequiresPower was set to false.

**Changelog**

:cl:
- tweak: Entities that are powered by APCs can now be designed so that they can electrocute you. Shocking!

### CODE REVIEW

File: Content.Server/Electrocution/ElectrocutionSystem.cs [hunk 1]
```diff
@@ -183,8 +183,18 @@ private void OnElectrifiedInteractUsing(EntityUid uid, ElectrifiedComponent elec
...
             var targets = new List<(EntityUid entity, int depth)>();
             GetChainedElectrocutionTargets(targetUid, targets);
-            if (!electrified.RequirePower)
+            if (!electrified.RequirePower || electrified.UsesApcPower)
             {
+                // Does it use APC power for its electrification check? Check if it's powered, and then
+                // attempt an electrocution if all the checks succeed.
+
+                if (electrified.UsesApcPower &&
+                    (!EntityManager.TryGetComponent(uid, out ApcPowerReceiverComponent power)
```
Comment (Fishfish458) [inline Content.Server/Electrocution/ElectrocutionSystem.cs:192]:
```suggestion
                    (!TryComp(uid, out ApcPowerReceiverComponent? power)
```

**Review by mirrorcult — APPROVED** [review]

**Comment by mirrorcult** [comment]
is this being added for something specific or just thought the feature would be nice to have in the future?

**Comment by vulppine** [comment]
> is this being added for something specific or just thought the feature would be nice to have in the future?

wires 👁️

**Comment by mirrorcult** [comment]
👁️

**Comment by vulppine** [comment]
having no way to do an electrification check via checking APC power really hindered the ability for a power wire to do a shock without disabling `requiresPower`, which isn't the correct way to go about it imo, so, i just

implemented it

wires soon™️

**Pull request merged**